ddl/partition.go

```diff
@@ -1232,19 +1232,41 @@ func (w *worker) onExchangeTablePartition(d *ddlCtx, t *meta.Meta, job *model.Jo
 		return ver, errors.Trace(err)
 	}
 
-	tempID := partDef.ID
-	// exchange table meta id
-	partDef.ID = nt.ID
-
 	if pt.TiFlashReplica != nil {
 		for i, id := range pt.TiFlashReplica.AvailablePartitionIDs {
-			if id == tempID {
-				pt.TiFlashReplica.AvailablePartitionIDs[i] = partDef.ID
+			if id == partDef.ID {
+				pt.TiFlashReplica.AvailablePartitionIDs[i] = nt.ID
 				break
 			}
 		}
 	}
 
+	if d.infoHandle != nil && d.infoHandle.IsValid() {
```

Comment:
could you make this as a func = {...}, and run it before finishDDLJob

Reply:
It does not make too much sense to me. I lost many argument in `finishDDLJob`. So i need to pass arguments. Then I need a `switch jobType` to handle things differently, which split the logic into two positions.

I do not see the need of moving, if other operations in `onExchangeTablePartition` will cancel the job, too.

Reply:
what if you `PutRuleBundles` succeed and the DDL failed and canceled in L1284?

Reply:
It is moved to the line before `updateSchemaVersion`. Code still works after swapping ID.